pkg/backup/backup.go

```diff
@@ -402,37 +415,71 @@ func (kb *kubernetesBackupper) BackupWithResolvers(
 	}()
 
 	backedUpGroupResources := map[schema.GroupResource]bool{}
+	// Maps items in the item list from GR+NamespacedName to a slice of pointers to kubernetesResources
+	// We need the slice value since if the EnableAPIGroupVersions feature flag is set, there may
+	// be more than one resource to back up for the given item.
+	itemsMap := make(map[velero.ResourceIdentifier][]*kubernetesResource)
+	for i := range items {
+		key := velero.ResourceIdentifier{
+			GroupResource: items[i].groupResource,
+			Namespace:     items[i].namespace,
+			Name:          items[i].name,
+		}
+		itemsMap[key] = append(itemsMap[key], items[i])
+	}
 
-	for i, item := range items {
+	var itemBlock *BackupItemBlock
+
+	for i := range items {
 		log.WithFields(map[string]interface{}{
 			"progress":  "",
-			"resource":  item.groupResource.String(),
-			"namespace": item.namespace,
-			"name":      item.name,
+			"resource":  items[i].groupResource.String(),
+			"namespace": items[i].namespace,
+			"name":      items[i].name,
 		}).Infof("Processing item")
 
-		// use an anonymous func so we can defer-close/remove the file
-		// as soon as we're done with it
-		func() {
-			var unstructured unstructured.Unstructured
-
-			f, err := os.Open(item.path)
-			if err != nil {
-				log.WithError(errors.WithStack(err)).Error("Error opening file containing item")
-				return
+		// Skip if this item has already been added to an ItemBlock
+		if items[i].inItemBlock {
+			log.Infof("Skipping %s %s/%s because it's already in an ItemBlock", items[i].groupResource.String(), items[i].namespace, items[i].name)
```

Comment:
I don't think it would be misunderstood as excluded, since "it's already in an ItemBlock", but I think it might make sense to make it debug. Any time an ItemBlock is created with multiple items, those items will later get this message.

Info makes it easier for users to see from the logs what's happening, though. I could go either way on this. But I'll modify the text a bit in either case. Maybe "Not creating new ItemBlock for  %s %s/%s because it's already in an ItemBlock"

Reply:
I agree that this new message "Not creating new ItemBlock for %s %s/%s because it's already in an ItemBlock" is better.

And for the Info log vs. debug log, please decide it based on your own understanding, personally, I feel it more like a debug info.